src/deploy-tes-on-azure/Utility.cs

```diff
@@ -134,5 +144,49 @@ public static string GetFileContent(params string[] pathComponentsRelativeToAppB
 
         private static Stream GetBinaryFileContent(params string[] pathComponentsRelativeToAppBase)
             => typeof(Deployer).Assembly.GetManifestResourceStream($"deploy-tes-on-azure.{string.Join(".", pathComponentsRelativeToAppBase)}");
+
+        // borrowed from https://github.com/Azure/azure-libraries-for-net/blob/7d85e294e4e7280c3f74b1c41438e2f20bce2052/src/ResourceManagement/ResourceManager/ResourceNamer.cs
+        private class ResourceNamer
+        {
+            private readonly string randName;
+            private static readonly Random random = new();
+
+            public ResourceNamer(string name)
+            {
+                lock (random)
```

Comment:
why does this need a lock?  seems like all the locks can be removed

Reply:
I just copied the code from the ADAL-connected library we are retiring. The new MSAL-connected library uses a completely different style of generated names and I didn't feel this is the PR to introduce this kind of behavior change. I didn't question the code since it's functional, but you have a good point. I'll clean this up.